package.json

```diff
@@ -13,6 +13,8 @@
     "precommit": "lint-staged",
     "commitmsg": "commitlint -e $GIT_PARAMS",
     "commit": "git-cz",
+    "build:changelog": "lerna-changelog --from=$(git log --grep='chore(release): releasing component' --format='%H' | sed -n 2p) > tmp && cat CHANGELOG.md >> tmp && cat tmp > CHANGELOG.md && rm tmp",
```

Comment:
Injecting crazium.

`lerna-changelog` includes the `--from` flag which also works with commit ID's. So, in order to fetch all PR's between the last release and current one we do the following...

1. Fetch all git log ID's that include the msg `chore(release): releasing component` which we use when releasing through `lerna`.
2. Select the 2nd line, as the first one is obviously the current release.
3. Appending this directly to the changelog would make it show up at the bottom of the file. So instead, we put it into a `tmp file` and then append the `changelog` to that file instead. Then finally overwrite the `changelog` with the result and deleting the `tmp file`

75% crazium injected

Reply:
I don't follow why we can just only run `lerna-changelog`?

Reply:
Good question. I'll try to explain below.

> By default lerna-changelog will show all pull requests that have been merged since the latest tagged commit in the repository
([ref](https://github.com/lerna/lerna-changelog#usage))

So, running `lerna-changelog` without any args **after** a release will not print out anything since no PRs has been made. 

On the other hand, running it **before** releasing would include the PRs, but put them under `## Unreleased` and thus, not include the versioning.

Exampe attached from docs below:
```
## Unreleased (2018-05-24)

#### :bug: Bug Fix
* [#198](https://github.com/my-org/my-repo/pull/198) Avoid an infinite loop ([@helpful-hacker](https://github.com/helpful-hacker))

#### :house: Internal
* [#183](https://github.com/my-org/my-repo/pull/183) Standardize error messages ([@careful-coder](https://github.com/careful-coder))

#### Commiters: 2
- Helpful Hacker ([@helpful-hacker](https://github.com/helpful-hacker))
- [@careful-coder](https://github.com/careful-coder)
```

Reply:
We run it _before_ a release. So the process is

1. Generate changelog
2. Commit to release branch
3. Merge PR
4. Release

This ensures that the `CHANGELOG.md` is actually part of the npm package. More [here](https://github.com/commercetools/merchant-center-application-kit/blob/master/CONTRIBUTING.md#cutting-a-release).

Reply:
Hmm very interesting. And lerna then updates the `.md` upon the actual release? Or do you do that manually after?

Reply:
Manually. We copy and paste it in. For us it works just fine. We don‘t want to release three times a day. So it‘s a 80/20 for how much automation we need.

Reply:
Ahh got it.. thanks a lot for sharing about your process on it.

Maybe having it fully automated is just a pipe dream of mine. Will sit down and rethink...

cc @katmatt